### PR TITLE
Document remaining sd-journal and sd-hwdb functions

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: man/sd_hwdb_get.xml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  <refsynopsisdiv>
+    <funcsynopsis>
+      <funcsynopsisinfo>#include &lt;systemd/sd-hwdb.h&gt;</funcsynopsisinfo>
+
+      <funcprototype>
+        <funcdef>int <function>sd_hwdb_get</function></funcdef>
+        <paramdef>sd_hwdb* <parameter>hwdb</parameter></paramdef>
+        <paramdef>const char* <parameter>modalias</parameter></paramdef>
+        <paramdef>const char* <parameter>key</parameter></paramdef>
+        <paramdef>const char** <parameter>value</parameter></paramdef>
```
Comment (daandemeyer) [inline man/sd_hwdb_get.xml]:
```suggestion
        <paramdef>sd_hwdb *<parameter>hwdb</parameter></paramdef>
        <paramdef>const char *<parameter>modalias</parameter></paramdef>
        <paramdef>const char *<parameter>key</parameter></paramdef>
        <paramdef>const char **<parameter>value</parameter></paramdef>
```

Comment (daandemeyer) [reply man/sd_hwdb_get.xml]: Parameters can have a right-aligned pointer right?

File: man/sd_hwdb_get.xml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>int <function>sd_hwdb_seek</function></funcdef>
+        <paramdef>sd_hwdb* <parameter>hwdb</parameter></paramdef>
+        <paramdef>const char* <parameter>modalias</parameter></paramdef>
```
Comment (daandemeyer) [inline man/sd_hwdb_get.xml]:
```suggestion
        <paramdef>sd_hwdb *<parameter>hwdb</parameter></paramdef>
        <paramdef>const char *<parameter>modalias</parameter></paramdef>
```

File: man/sd_hwdb_get.xml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>int <function>sd_hwdb_enumerate</function></funcdef>
+        <paramdef>sd_hwdb* <parameter>hwdb</parameter></paramdef>
+        <paramdef>const char** <parameter>key</parameter></paramdef>
+        <paramdef>const char** <parameter>value</parameter></paramdef>
```
Comment (daandemeyer) [inline man/sd_hwdb_get.xml]:
```suggestion
        <paramdef>sd_hwdb *<parameter>hwdb</parameter></paramdef>
        <paramdef>const char **<parameter>key</parameter></paramdef>
        <paramdef>const char **<parameter>value</parameter></paramdef>
```

File: man/sd_hwdb_get.xml [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    <parameter>modalias</parameter>, the combination of all matching key-value pairs is used. See
+    <citerefentry><refentrytitle>hwdb</refentrytitle><manvolnum>7</manvolnum></citerefentry> for
+    details.</para>
+  </refsect1>
+
+  <refsect1>
+    <title>Return Value</title>
+
+    <para>On success, <function>sd_hwdb_get()</function> and <function>sd_hwdb_seek()</function> return 0 or
+    a positive integer. On failure, they return a negative errno-style error code.</para>
```
Comment (daandemeyer) [inline man/sd_hwdb_get.xml]:
```suggestion
    <para>On success, <function>sd_hwdb_get()</function> and <function>sd_hwdb_seek()</function> return a 
    non-negative integer. On failure, they return a negative errno-style error code.</para>
```

File: man/sd_hwdb_get.xml [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+    <para><function>sd_hwdb_enumerate()</function> returns 1 if another key-value pair was found, and 0 if
```
Comment (daandemeyer) [inline man/sd_hwdb_get.xml]:
```suggestion
    <para><function>sd_hwdb_enumerate()</function> returns a positive integer if another key-value pair was found or zero if
```

File: man/sd_hwdb_new.xml [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+      <funcsynopsisinfo>#include &lt;systemd/sd-hwdb.h&gt;</funcsynopsisinfo>
+
+      <funcprototype>
+        <funcdef>int <function>sd_hwdb_new</function></funcdef>
+        <paramdef>sd_hwdb **<parameter>hwdb</parameter></paramdef>
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>sd_hwdb* <function>sd_hwdb_ref</function></funcdef>
+        <paramdef>sd_hwdb* <parameter>hwdb</parameter></paramdef>
```
Comment (daandemeyer) [inline man/sd_hwdb_new.xml]:
```suggestion
        <paramdef>sd_hwdb *<parameter>hwdb</parameter></paramdef>
```

File: man/sd_hwdb_new.xml [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>sd_hwdb* <function>sd_hwdb_unref</function></funcdef>
+        <paramdef>sd_hwdb* <parameter>hwdb</parameter></paramdef>
```
Comment (daandemeyer) [inline man/sd_hwdb_new.xml]:
```suggestion
        <paramdef>sd_hwdb *<parameter>hwdb</parameter></paramdef>
```

File: man/sd_hwdb_new.xml [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    <title>Description</title>
+
+    <para><function>sd_hwdb_new()</function> creates a new hwdb object to access the binary hwdb
+    database. Upon initialization, the file containing the binary representation of the hardware database is
+    located and opened. The new object is returned in <parameter>hwdb</parameter>.</para>
+
+    <para>The <parameter>hwdb</parameter> object is reference counted. <function>sd_hwdb_ref()</function> and
+    <function>sd_hwdb_unref()</function> may be used to get a new reference or destroy an existing reference
+    to an object. The caller must dispose of the reference acquired with <function>sd_hwdb_new()</function>
+    by calling <function>sd_hwdb_unref()</function> when done with the object.</para>
```
Comment (daandemeyer) [inline man/sd_hwdb_new.xml:56]:
```suggestion
    by calling <function>sd_hwdb_unref()</function> when he has finished using the object.</para>
```

File: man/sd_hwdb_new.xml [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+    <citerefentry><refentrytitle>sd_hwdb_seek</refentrytitle><manvolnum>3</manvolnum></citerefentry>,
+    <citerefentry><refentrytitle>sd_hwdb_get</refentrytitle><manvolnum>3</manvolnum></citerefentry>, and
+    <citerefentry><refentrytitle>sd_hwdb_enumerate</refentrytitle><manvolnum>3</manvolnum></citerefentry> to
+    access entries.</para>
+  </refsect1>
+
+  <refsect1>
+    <title>Return Value</title>
+
+    <para>On success, <function>sd_hwdb_new()</function> returns 0 or a positive integer. On
```
Comment (daandemeyer) [inline man/sd_hwdb_new.xml]:
```suggestion
    <para>On success, <function>sd_hwdb_new()</function> returns a non-negative integer. On
```

File: man/sd_hwdb_new.xml [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+    <refsect2>
+      <title>Errors</title>
+
+      <para>Returned errors may indicate the following problems:</para>
+
+      <variablelist>
+        <varlistentry>
+          <term><constant>-ENOENT</constant></term>
+
+          <listitem><para>Binary hwdb could not be located. See
```
Comment (daandemeyer) [inline man/sd_hwdb_new.xml]:
```suggestion
          <listitem><para>The binary hwdb could not be located. See
```

File: man/sd_journal_print.xml [hunk 11]
```diff
@@ -136,11 +187,20 @@ sd_journal_send("MESSAGE=Hello World, this is PID %lu!", (unsigned long) getpid(
...
+    </para>
+
+    <para><function>sd_journal_print_with_location</function>,
+    <function>sd_journal_printv_with_location</function>, <function>sd_journal_send_with_location</function>,
+    <function>sd_journal_sendv_with_location</function>, and
+    <function>sd_journal_perror_with_location</function> are similar to their counterparts without
+    <literal>_with_location</literal>, but accept additional parameters to explicitly set the source file
+    name, function, and line. Those arguments must contain valid journal entries including the variable name,
+    e.g. <literal>CODE_FILE=src/foo.c</literal>, <literal>CODE_LINE=666</literal>,
+    <literal>CODE_FUNC=myfunc</literal>. Those variants are primarily useful when writing custom wrappers,
```
Comment (daandemeyer) [inline man/sd_journal_print.xml]:
```suggestion
    <literal>CODE_FUNC=myfunc</literal>. These variants are primarily useful when writing custom wrappers,
```

File: man/sd_hwdb_get.xml (hunk 1 above)
Comment (keszybz) [reply man/sd_hwdb_get.xml]: For some reason, I thought that docbook reformats this, and the alignment used in xml doesn't matter. It does some reformatting, but actually doesn't move the pointer symbol. So yeah, this should be right-aligned.

File: man/sd_hwdb_new.xml (hunk 8 above)
Comment (keszybz) [reply man/sd_hwdb_new.xml:56]: Gender-neutral forms are preferred. Also I think the original text here is OK.

File: man/sd_hwdb_get.xml [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+    <title>Examples</title>
+
+    <example>
+      <title>Look up hwdb entries for a specific match string</title>
+
+      <programlisting>
+#include &lt;stdio.h&gt;
+#include &lt;sd-hwdb.h&gt;
+
+int lookup(char *match) {
```
Comment (poettering) [inline man/sd_hwdb_get.xml]: to make the example more real-lifey, maybe beef it up to do this:

```c
int lookup_usb_properties(uint16_t vid, uint16_t pid) {
        char match[15];
        /* Search fields matching by USB vendor/product IDs */
        snprintf(match, "usb:v%04Xp%04X", vid, pid);
        …    
}
```

(just an idea)

File: man/sd_hwdb_new.xml [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+    <refsect2>
+      <title>Errors</title>
+
+      <para>Returned errors may indicate the following problems:</para>
+
+      <variablelist>
+        <varlistentry>
+          <term><constant>-ENOENT</constant></term>
+
+          <listitem><para>The binary hwdb could not be located. See
```
Comment (poettering) [inline man/sd_hwdb_new.xml]: "hwdb" → "hardware database file"?

**Comment by keszybz** [comment]
Updated with the suggested changes.

File: man/hwdb-usb-device.c [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+#include <stdint.h>
+#include <sd-hwdb.h>
+
+int print_usb_properties(uint16_t vid, uint16_t pid) {
+  char match[15];
+  sd_hwdb *hwdb;
+  const char *key, *value;
+  int r;
+
+   /* Search fields matching by USB vendor/product IDs */
```
Comment (poettering) [inline man/hwdb-usb-device.c]: indentation slightly off

Comment (keszybz) [reply man/hwdb-usb-device.c]: Fixed now. I changed the comment a bit too.

**Pull request merged**